plugins/module_utils/mh/module_helper_dest_file.py

```diff
@@ -0,0 +1,213 @@
+from ansible_collections.community.general.plugins.module_utils.mh.module_helper import ModuleHelper
+from ansible_collections.community.general.plugins.module_utils.mh.exceptions import ModuleHelperException
+from ansible_collections.community.general.plugins.module_utils.mh.deco import (
+    cause_changes,
+    check_mode_skip)
+
+
+class DestNotExists(ModuleHelperException):
+    pass
+
+
+class ParentNotWriteable(ModuleHelperException):
+    pass
+
+
+class DestNotReadable(ModuleHelperException):
+    pass
+
+
+class DestNotWriteable(ModuleHelperException):
+    pass
+
+
+class DestNotRegularFile(ModuleHelperException):
+    pass
+
+
+class CantCreateBackup(ModuleHelperException):
+    pass
+
+
+def check_if_dest_exists(dest):
+    # type: (str) -> bool
+    try:
+        os.stat(dest)
+    except FileNotFoundError:
+        return False
+    except PermissionError:
+        pass
+    return True
+
+
+def check_if_parent_is_writeable(dest):
+    # type: (str) -> bool
+    if os.access(os.path.dirname(dest), os.W_OK):
+        return True
+    return False
+
+
+def check_if_dest_is_readable(dest):
+    # type: (str) -> bool
+    if os.access(dest, os.R_OK):
+        return True
+    return False
+
+
+def check_if_dest_is_writeable(dest):
+    # type: (str) -> bool
+    if os.access(dest, os.W_OK):
+        return True
+    return False
+
+
+def check_if_dest_is_regular_file(dest):
+    # type: (str) -> bool
+    if os.path.isfile(dest):
+        return True
+    return False
+
+
+def dest_file_sanity_check(dest, create=False, backup=False):
+    # type: (str, bool, bool) -> bool
+    """
+    Ensure that the destination exist and is readable or if not exists and
+    'create' is True, ensure that the parent of the destination is writeable. If
+    'backup' is True, also ensure that the backup file can be created.
+
+    Return `True` if the file will be created else `False`.
+    """
+    exists = check_if_dest_exists(dest)
+    writeable_parent = check_if_parent_is_writeable(dest)
+    if not create and not exists:
+        raise DestNotExists(msg="{0} does not not exist !".format(dest))
+    if create and not exists and not writeable_parent:
+        raise ParentNotWriteable(msg="Can't create {0}, parent directory is not writeable !".format(dest))
+    if exists and not check_if_dest_is_readable(dest):
+        raise DestNotReadable(msg="Destination {0} is not readable !".format(dest))
+    if exists and not check_if_dest_is_writeable(dest):
+        raise DestNotWriteable(msg="Destination {0} is not writeable !".format(dest))
+    if exists and not check_if_dest_is_regular_file(dest):
+        raise DestNotRegularFile(msg="Destination {0} is not a regular file !".format(dest))
+    if exists and backup and not writeable_parent:
+        raise CantCreateBackup(msg="Can't create backup file for {0} because parrent dir is not writeable !".format(dest))
+    return (not exists and create)
+
+
+class DestFileModuleHelper(ModuleHelper):
```

Comment:
Your right, it seem better to implement this as Mixins.

Otherwise, i will explore the other ideas. I need to try to implement them to get cleared idea of what it might look like. I keep in mind my first goal : make modules that will be able to ensure a specifics values are present or absent from a JSON, YAML or TOML file, easily, simply with only one task. I think the first step to put me on the way that you gave me it to make the `DataMergeUtils` also available as a filter.